Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/common/WasbUri.java

```diff
@@ -43,29 +44,26 @@
                     + "(:(?<port>[0-9]+))?(/(?<path>.*))?$",
             Pattern.CASE_INSENSITIVE);
 
-    private final URI rawUri;
     private final LaterInit<String> container = new LaterInit<>();
     private final LaterInit<String> storageAccount = new LaterInit<>();
     private final LaterInit<String> endpointSuffix = new LaterInit<>();
-    private final LaterInit<String> relativePath = new LaterInit<>();
+    private final LaterInit<String> path = new LaterInit<>();
 
     private WasbUri(final URI rawUri) {
-        this.rawUri = rawUri;
-    }
-
-    public URI getRawUri() {
-        return rawUri;
+        super(rawUri);
     }
 
+    @Override
     public URI getUri() {
         return URI.create(String.format("wasbs://%s@%s.blob.%s%s",
-                getContainer(), getStorageAccount(), getEndpointSuffix(), getRelativePath()));
+                getContainer(), getStorageAccount(), getEndpointSuffix(), getPath()));
     }
 
+    @Override
     public URL getUrl() {
         try {
             return URI.create(String.format("https://%s.blob.%s/%s%s",
-                    getStorageAccount(), getContainer(), getEndpointSuffix(), getRelativePath())).toURL();
+                    getStorageAccount(), getContainer(), getEndpointSuffix(), getPath())).toURL();
```

Comment:
I suppose the order should be `getStorageAccount(), getEndpointSuffix(), getContainer(), getPath()`. Could you also add UT for class `WasbUri`

Reply:
Good catch!